airflow/ui/src/pages/TaskInstance/Logs.tsx

```diff
@@ -77,26 +84,47 @@ export const Logs = () => {
 
   return (
     <Box p={2}>
-      <HStack justifyContent="space-between" mb={2}>
-        {taskInstance === undefined || tryNumber === undefined || taskInstance.try_number <= 1 ? (
-          <div />
-        ) : (
-          <TaskTrySelect
-            onSelectTryNumber={onSelectTryNumber}
-            selectedTryNumber={tryNumber}
-            taskInstance={taskInstance}
-          />
-        )}
-        <Button aria-label={wrap ? "Unwrap" : "Wrap"} bg="bg.panel" onClick={toggleWrap} variant="outline">
-          {wrap ? "Unwrap" : "Wrap"}
-        </Button>
-      </HStack>
+      <TaskLogHeader
+        onSelectTryNumber={onSelectTryNumber}
+        taskInstance={taskInstance}
+        toggleFullscreen={toggleFullscreen}
+        toggleWrap={toggleWrap}
+        tryNumber={tryNumber}
+        wrap={wrap}
+      />
       <ErrorAlert error={error ?? logError} />
       <Skeleton />
       <ProgressBar size="xs" visibility={isLoading || isLoadingLogs ? "visible" : "hidden"} />
       <Code overflow="auto" py={3} textWrap={wrap ? "pre" : "nowrap"}>
         <VStack alignItems="flex-start">{data.parsedLogs}</VStack>
       </Code>
+      <Dialog.Root onOpenChange={onOpenChange} open={fullscreen} scrollBehavior="inside" size="full">
+        <Dialog.Content backdrop>
+          <Dialog.Header>
+            <Heading size="xl">{taskId}</Heading>
+            <TaskLogHeader
+              isFullscreen
+              onSelectTryNumber={onSelectTryNumber}
+              taskInstance={taskInstance}
+              toggleFullscreen={toggleFullscreen}
+              toggleWrap={toggleWrap}
+              tryNumber={tryNumber}
+              wrap={wrap}
+            />
+          </Dialog.Header>
+
+          <Dialog.CloseTrigger />
+
+          <Dialog.Body>
+            <ErrorAlert error={error ?? logError} />
+            <Skeleton />
+            <ProgressBar size="xs" visibility={isLoading || isLoadingLogs ? "visible" : "hidden"} />
+            <Code overflow="auto" py={3} textWrap={wrap ? "pre" : "nowrap"}>
+              <VStack alignItems="flex-start">{data.parsedLogs}</VStack>
+            </Code>
```

Comment:
This should be factorized, maybe inside a `LogContent` Component. Inside that same `Logs` folder. Because for now it repeats, and it will grow bigger and bigger with log parsing, and colors, and groups. We will need it at some point.

Maybe even bring in the skeleton and progressbar.

Reply:
Thanks, refactored to `TaskLogContent.tsx` for reuse inside Logs folder.